pkg/capabilities/capabilities.go

```diff
@@ -113,7 +113,7 @@ type BaseCapability interface {
 }
 
 type TriggerExecutable interface {
-	RegisterTrigger(ctx context.Context, callback chan<- CapabilityResponse, request CapabilityRequest) error
+	RegisterTrigger(ctx context.Context, request CapabilityRequest) (<-chan CapabilityResponse, error)
```

Comment:
We discussed changing CapabilityResponse to TriggerEvent here. Maybe it make sense to include in this PR right away?